base/src/com/google/idea/blaze/base/settings/Blaze.java

```diff
@@ -72,6 +78,38 @@ public static ProjectType getProjectType(@Nullable Project project) {
     return blazeImportSettings.getProjectType();
   }
 
+  /**
+   * This variant allows us to enable and disable Query Sync for already imported project.
+   * com.google.idea.blaze.base.settings.Blaze#getProjectType(com.intellij.openapi.project.Project) is called quite often
+   * so we cannot reload project view from for every of such call.
+   * This is why we have this special case to make sure that Sync respects project view selection if there is any.
+   */
+  public static ProjectType getProjectTypeBeforeSync(@Nonnull Project project) {
```

Comment:
I've been also looking for a way to jump between qs/as on the fly, thanks for submitting this PR! Can we add some javadoc here to reflect somthing like `returns the project type with project view file lookup`? and maybe make the name more matching the action it does rather than a point it code where it should be called?

Reply:
There are 15 places where `Blaze.getProjectType*(project) == ProjectType.QUERY_SYNC` check is performed. So I wanted to have a special case for sync calls only. This is why name reflects this.

I don't really know how frequent the rest of places are hitting this method and saw it called from action apperance update, which cannot afford to reload file for sure.

So instead of having "get type with refresh" I'd rather keep "get type for sync and make sure the proper sync mode is used". Also we are updating `BazelImportSettings` here and it is ok to update those on sync but for the rest of ops I'd like to avoid this.

Reply:
I can rename it to `getSelectedProjectTypeBeforeSync` or so. The fact that `BazelImportSetings` are updated is important yet subtle here. Before this change it was set once during project creation and not touched, for whatever reason.

Overall `Blaze.getProjectType*(project) == ProjectType.QUERY_SYNC` is someway out of control... It is peppered over the code quite randomly and I'm a little bit worried about it causing trouble in the future even for cases beyond sync.

Reply:
`getRefreshedProjectType` or `getNotCachedBeforeSync` is another option... But then it will be a dilemma where to call those :)